src/rail/core/data.py

```diff
@@ -296,6 +304,10 @@ def _write_chunk(cls, data, fileObj, groups, start, end, **kwargs):
     def _finalize_write(cls, data, fileObj, **kwargs):
         return data.finalizeHdf5Write(fileObj)
 
+    @classmethod
+    def _validate_data(cls, data):
+        assert isinstance(data, qp.Ensemble)
```

Comment:
Yeah, this looks pretty straightforward. I'm guessing that we would want to add some logging or print to terminal before the assert statement. 
```suggestion
        if not isinstance(data, qp.Ensemble):
            print(f"Expected `data` to be a `qp.Ensemble`, but {type(data)} was provided. Perhaps you meant to use `TableHandle`?")
            assert isinstance(data, qp.Ensemble)
```